lib/experimental/html/class-wp-html-tag-processor.php

```diff
@@ -1280,24 +1312,37 @@ private function apply_attributes_updates() {
 			$tail_delta = 0;
 
 			foreach ( $this->lexical_updates as $diff ) {
-				$update_head = $bookmark->start >= $diff->start;
-				$update_tail = $bookmark->end >= $diff->start;
+				$bookmark_start_is_after_diff_start = $bookmark->start >= $diff->start;
+				$bookmark_end_is_after_diff_end     = $bookmark->end >= $diff->start;
+
+				if ( $bookmark_start_is_after_diff_start ) {
+					$bookmark_end_is_before_diff_end = $bookmark->end < $diff->end;
+					if ( $bookmark_end_is_before_diff_end ) {
+						// The bookmark is fully contained within the diff. We need to invalidate it.
+						$this->release_bookmark( $bookmark_name );
+					}
+				}
 
-				if ( ! $update_head && ! $update_tail ) {
+				if ( ! $bookmark_start_is_after_diff_start && ! $bookmark_end_is_after_diff_end ) {
 					break;
 				}
 
 				$delta = strlen( $diff->text ) - ( $diff->end - $diff->start );
 
-				if ( $update_head ) {
+				if ( $bookmark_start_is_after_diff_start ) {
 					$head_delta += $delta;
 				}
 
-				if ( $update_tail ) {
+				if ( $bookmark_end_is_after_diff_end ) {
 					$tail_delta += $delta;
 				}
 			}
 
+			// Did we end up invalidating the bookmark?
```

Comment:
Just noting this assumes that bookmarks are sorted and no processing is required for all further bookmarks – is that right?

Reply:
Ah, good spot! That should be `continue`, not `break` 😅

Reply:
https://github.com/WordPress/gutenberg/pull/47068/commits/9bfaab8b4603f7b14456bde74b06f94339204ce8